mm2src/coins/eth.rs

```diff
@@ -2791,13 +2804,62 @@ async fn sign_raw_eth_tx(coin: &EthCoin, args: &SignEthTransactionParams) -> Raw
             })
             .map_to_mm(|err| RawTransactionError::TransactionError(err.get_plain_text_format()))
         },
+        EthPrivKeyPolicy::WalletConnect { .. } => {
+            // NOTE: doesn't work with wallets that doesn't support `eth_signTransaction`.
+            // e.g Metamask
+            let wc = {
+                let ctx = MmArc::from_weak(&coin.ctx).expect("No context");
+                WalletConnectCtx::from_ctx(&ctx)
+                    .expect("TODO: handle error when enable kdf initialization without key.")
+            };
+            let my_address = coin
+                .derivation_method
+                .single_addr_or_err()
+                .await
+                .mm_err(|e| RawTransactionError::InternalError(e.to_string()))?;
+            let address_lock = coin.get_address_lock(my_address.to_string()).await;
+            let _nonce_lock = address_lock.lock().await;
+            let pay_for_gas_option = if let Some(ref pay_for_gas) = args.pay_for_gas {
+                pay_for_gas.clone().try_into()?
+            } else {
+                // use legacy gas_price() if not set
+                info!(target: "sign-and-send", "get_gas_price…");
+                let gas_price = coin.get_gas_price().await?;
+                PayForGasOption::Legacy(LegacyGasPrice { gas_price })
+            };
+            let (nonce, _) = coin
+                .clone()
+                .get_addr_nonce(my_address)
+                .compat()
+                .await
+                .map_to_mm(RawTransactionError::InvalidParam)?;
+
+            info!(target: "sign-and-send", "WalletConnect signing and sending tx…");
+            let (signed_tx, _) = coin
+                .wc_sign_tx(&wc, WcEthTxParams {
+                    my_address,
+                    gas_price: pay_for_gas_option.get_gas_price(),
+                    action,
+                    value,
+                    gas: args.gas_limit,
+                    data: &data,
+                    nonce,
+                    chain_id: coin.chain_id as u8,
```

Comment:
Some chains have a chain id that will not fit into u8 like avax, how will it work with WalletConnect, I see that it is supported. Maybe @smk762 should test it after we find how to handle it.

Reply:
I didnt transact with AVAX, but I did have a successful activation. withdraw/swap tests were done with MATIC, USDC-PLG20 and IRIS. Happy to recheck with AVAX when ready.

Reply:
updated chain to u64 and removed castings to u8

Reply:
> Happy to recheck with AVAX when ready.

It's ready now but I don't think there is a need for recheck as I checked the WalletConnect docs and they use the same`43114` chain id.